### PR TITLE
update fee mw docs, add formating, fix typos, increase readability

### PR DESCRIPTION
**Pull request by tmsdkeys**
## Description

I've updated the fee middleware docs to correct some typos, nits. Also fixed the ordering in the navigation and made sure the section titles are consistent when referencing. Added some diagrams. Fixed some syntax highlighting and added formatting to the docs to increase readability.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [X] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [X] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/middleware/ics29-fee/overview.md [hunk 1]
```diff
@@ -8,17 +8,29 @@ Learn about what the Fee Middleware module is, and how to build custom modules t
...
+
+> To achieve the stated requirements, the **fee middleware module has two main groups of functionality**:
+
+- Registering of relayer addresses associated with each party involved in relaying the packet on the source chain. This registration process can be automated on start up of relayer infrastructure and happens only once, not every packet flow.
+
+  This is described in the [fee distribution section](../ics29-fee/fee-distribution.md).
+
+- Escrowing fees by any party which will be paid out to each rightful party on completion of the packet lifecycle.
+
+  This is described in the [fee messages section](../ics29-fee/msgs.md).
```
Comment (crodriguezvega) [inline docs/middleware/ics29-fee/overview.md:31]:
```suggestion
  This is described in the [Fee messages section](../ics29-fee/msgs.md).
```

**Review by AdityaSripal — COMMENTED** [review]
Looks great! I personally like capitalizing each word in the header and in phrases @crodriguezvega . Not a huge deal, but we should decide on a standard and stick to it

**Review by AdityaSripal — APPROVED** [review]

**Comment by AdityaSripal** [comment]
Also excellent diagrams!

File: docs/middleware/ics29-fee/fee-distribution.md [hunk 2]
```diff
@@ -42,27 +46,31 @@ type MsgRegisterCounterpartyPayee struct {
...
+>
+> - `PortId` is invalid (see [24-host naming requirements](https://github.com/cosmos/ibc/blob/master/spec/core/ics-024-host-requirements/README.md#paths-identifiers-separators).
+> - `ChannelId` is invalid (see [24-host naming requirements](https://github.com/cosmos/ibc/blob/master/spec/core/ics-024-host-requirements/README.md#paths-identifiers-separators)).
+> - `Relayer` is an invalid address (see [Cosmos SDK Addresses](https://github.com/cosmos/cosmos-sdk/blob/main/docs/basics/accounts.md#Addresses)).
+> - `CounterpartyPayee` is empty.
 
 See below for an example CLI command:
 
 ```
 simd tx ibc-fee register-counterparty-payee transfer channel-0 cosmos1rsp837a4kvtgp2m4uqzdge0zzu6efqgucm0qdh osmo1v5y0tz01llxzf4c2afml8s3awue0ymju22wxx2 --from cosmos1rsp837a4kvtgp2m4uqzdge0zzu6efqgucm0qdh
```
[the remark this answers is not in the document]
Comment (tmsdkeys) [reply docs/middleware/ics29-fee/fee-distribution.md:59]: sure, will implement!

File: docs/middleware/ics29-fee/fee-distribution.md [hunk 3]
```diff
@@ -80,12 +88,12 @@ type MsgRegisterPayee struct {
...
-- `Relayer` is an invalid address (see [Cosmos SDK Addresses](https://github.com/cosmos/cosmos-sdk/blob/main/docs/basics/accounts.md#Addresses)).
-- `Payee` is an invalid address (see [Cosmos SDK Addresses](https://github.com/cosmos/cosmos-sdk/blob/main/docs/basics/accounts.md#Addresses)).
+> This message is expected to fail if:
+>
+> - `PortId` is invalid (see [24-host naming requirements](https://github.com/cosmos/ibc/blob/master/spec/core/ics-024-host-requirements/README.md#paths-identifiers-separators).
+> - `ChannelId` is invalid (see [24-host naming requirements](https://github.com/cosmos/ibc/blob/master/spec/core/ics-024-host-requirements/README.md#paths-identifiers-separators)).
+> - `Relayer` is an invalid address (see [Cosmos SDK Addresses](https://github.com/cosmos/cosmos-sdk/blob/main/docs/basics/accounts.md#Addresses)).
+> - `Payee` is an invalid address (see [Cosmos SDK Addresses](https://github.com/cosmos/cosmos-sdk/blob/main/docs/basics/accounts.md#Addresses)).
 
 See below for an example CLI command:
```
[the remark this answers is not in the document]
Comment (tmsdkeys) [reply docs/middleware/ics29-fee/fee-distribution.md:98]: will do

**Comment by tmsdkeys** [comment]
@crodriguezvega thanks for your review! I agree with your nits on the diagrams, I could check with the design team if they can update these changes.

That is, if we come to the agreement to use the design diagrams vs Excalidraw. Personally, I don't mind using Excalidraws as it concerns the developer documentation. However, for other external comms we will likely keep using designed diagrams so maybe for overall consistency it's useful to use the design diagrams everywhere? What do others think here?

Also, re the headings: I have no true preference for capitalising every word or not but agree we should come to conensus and keep it consistent.

+ yes, I've run the docs locally and the diagrams look quite good imo in the version built with vuepress whereas in regular md files they look comically large

File: docs/middleware/ics29-fee/overview.md [hunk 4]
```diff
@@ -8,17 +8,29 @@ Learn about what the Fee Middleware module is, and how to build custom modules t
...
 
-## Concepts 
+## Concepts
 
-ICS29 fee payments in this middleware design are built on the assumption that sender chains are the source of incentives — the chain on which packets are sent is the same chain that fee distribution to relayer operators takes place. Therefore, the middleware enables the registering of addresses associated with each party involved in relaying the packet on the source chain, and the escrowing of fees by any party which will be paid out to each party on completion of the packet lifecycle. This registration process can be automated on start up of relayer infrastructure.
+ICS29 fee payments in this middleware design are built on the assumption that sender chains are the source of incentives — the chain on which packets are is the chain that distributes fees to relayer operators. However, as part of the IBC packet flow, messages have to be submitted on both sender and destination chains. This introduces the requirement of a mapping of relayer operator's addresses on both chains.
```
Comment (colin-axner) [inline docs/middleware/ics29-fee/overview.md:21]:
```suggestion
ICS29 fee payments in this middleware design are built on the assumption that sender chains are the source of incentives — the chain on which packets are incentivized is the chain that distributes fees to relayer operators. However, as part of the IBC packet flow, messages have to be submitted on both sender and destination chains. This introduces the requirement of a mapping of relayer operator's addresses on both chains.
```

**Review by colin-axner — APPROVED** [review]
Thank you! Great work :mechanical_arm:

**Comment by crodriguezvega** [comment]
>That is, if we come to the agreement to use the design diagrams vs Excalidraw. Personally, I don't mind using Excalidraws as it concerns the developer documentation. However, for other external comms we will likely keep using designed diagrams so maybe for overall consistency it's useful to use the design diagrams everywhere? What do others think here?

@tmsdkeys I don't have a strong preference for using only one style or mix different diagram styles. I just mentioned it to see what others would think. So if it's easier to keep the diagrams from the design team, that's perfectly fine for me.

**Pull request merged**